validphys2/src/validphys/plotoptions/plottingoptions.py

```diff
@@ -0,0 +1,105 @@
+from validphys.plotoptions import kintransforms, labelers, resulttransforms
+from validphys.plotoptions.utils import get_subclasses
+
+default_labels = ('idat', 'k1', 'k2', 'k3')
+
+
+labeler_functions = get_functions(labelers)
+transform_functions = get_subclasses(kintransforms, kintransforms.Kintransform)
+result_functions = get_functions(resulttransforms)
+
+
+ResultTransformations = enum.Enum('ResultTransformations', list(result_functions.keys()))
+TransformFunctions = enum.Enum('TransformFunctions', list(transform_functions.keys()))
+
+
+class Scale(enum.Enum):
+    linear = enum.auto()
+    log = enum.auto()
+    symlog = enum.auto()
+
+
+@dataclasses.dataclass
+class PlottingOptions:
+    func_labels: dict = dataclasses.field(default_factory=dict)
+    dataset_label: typing.Optional[str] = None
+    experiment: typing.Optional[str] = None
+    nnpdf31_process: typing.Optional[str] = None
+    data_reference: typing.Optional[str] = None
+    theory_reference: typing.Optional[str] = None
+    process_description: typing.Optional[str] = None
+    y_label: typing.Optional[str] = None
+    x_label: typing.Optional[str] = None
+
+    kinematics_override: typing.Optional[TransformFunctions] = None
+
+    result_transform: typing.Optional[ResultTransformations] = None
+
+    # TODO: change this to x: typing.Optional[KinLabel] = None
+    # but this currently fails CI because some datasets have
+    # a kinlabel of $x_1$ or " "!!
+    x: typing.Optional[str] = None
+
+    # TODO: the old commondata uses x, the new uses plot_x
+    # the old commondata only allowed the x to be k1, k2, k3 or idat
+    plot_x: typing.Optional[str] = None
+
+    x_scale: typing.Optional[Scale] = None
+    y_scale: typing.Optional[Scale] = None
+
+    line_by: typing.Optional[list] = None
+    figure_by: typing.Optional[list] = None
+
+    extra_labels: typing.Optional[typing.Mapping[str, typing.List]] = None
+
+    # TODO: the old commondata saved this normalize key in a different way
+    # need to check it is equivalent in all dataset it appears before merging
+    normalize: typing.Optional[dict] = None
+
+    # The new commondata files might need to change some variables inside the plotting
+    # avoid doing it twice
+    already_digested: typing.Optional[bool] = False
+
+    def parse_figure_by(self):
+        if self.figure_by is not None:
+            for el in self.figure_by:
+                if el in labeler_functions:
+                    self.func_labels[el] = labeler_functions[el]
+
+    def parse_line_by(self):
+        if self.line_by is not None:
+            for el in self.line_by:
+                if el in labeler_functions:
+                    self.func_labels[el] = labeler_functions[el]
+
+    def parse_x(self):
+        if self.x is not None and self.x not in self.all_labels:
+            raise ValidationError(
+                f"The label {self.x} is not in the set of known labels {self.all_labels}"
+            )
+
```

Comment:
Shouldn't [results_transform](https://github.com/NNPDF/nnpdf/blob/master/validphys2/src/validphys/plotoptions/resulttransforms.py) also be parsed here? Some datasets require this for the plotting.

Reply:
Results transforms hasn't changed (at least I think it hasn't) for any of the new datasets so it can be taken raw from the metadata (like 49).

This is akin to what was done before, only then it was taken from the plotting.

As I said, _I think_ it hasn't failed, I went over the plots and didn't see anything broken, but there were many plots so the changes that I missed something are high.

Reply:
Could you share the link to the plots? I can try to have a look as well.

Reply:
I thought I had but I cannot find the links so I clearly didn't.

Atlas: https://vp.nnpdf.science/0YL4eaPTT7eR3wM9IQwj5w==
BCDMS: https://vp.nnpdf.science/E0IDIgWFRF6tvdFk3pD0mA==
Chorus: https://vp.nnpdf.science/wCFRKBNsSA2U2O_6Sa75Zw==
CMS: https://vp.nnpdf.science/oSZLrPg3Tyyf-i2mE33G-Q==
HERA: https://vp.nnpdf.science/UHU-TYLJQCuE-8lRHe8Aog==
FTDY: https://vp.nnpdf.science/Toy_r6uFRm-h1oUFEZF6hw==
NMC:  https://vp.nnpdf.science/-2VyNN3CTHWnb26fUQkUJQ==
Nutev: https://vp.nnpdf.science/YPAQHvMtTeyBuNq12nl6RQ==
Slac: https://vp.nnpdf.science/chwFM_lJR025vREJ1I9VPQ==

I haven't been able to upload tevatron because the INFN has decided that I was connecting many times (I just looped over all the reports) and I'm banned for a few minutes or hours xD
LHCb I didn't do because those datasets were not ported automatically.

Reply:
Looks good! I went through the reports and didn't notice anything wrong.